packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -302,63 +304,156 @@ List<Plugin> findPlugins(FlutterProject project) {
   }
   packages.forEach((String name, Uri uri) {
     final Uri packageRoot = uri.resolve('..');
-    final Plugin plugin = _pluginFromPubspec(name, packageRoot);
+    final Plugin plugin = _pluginFromPackage(name, packageRoot);
     if (plugin != null) {
       plugins.add(plugin);
     }
   });
   return plugins;
 }
 
-/// Writes the .flutter-plugins and .flutter-plugins-dependencies files based on the list of plugins.
-/// If there aren't any plugins, then the files aren't written to disk.
+  /// Filters [plugins] to those supported by [platformKey].
+  List<Map<String, dynamic>> _filterPluginsByPlatform(List<Plugin>plugins, String platformKey) {
+    final Iterable<Plugin> platformPlugins = plugins.where((Plugin p) {
+      return p.platforms.containsKey(platformKey);
+    });
+
+    final List<Map<String, dynamic>> list = <Map<String, dynamic>>[];
+    for (final Plugin plugin in platformPlugins) {
+      list.add(<String, dynamic>{
+        'name': plugin.name,
+        'path': fsUtils.escapePath(plugin.path),
+      });
+    }
+    return list;
+  }
+
+/// Writes the .flutter-plugins-dependencies file based on the list of plugins.
+/// If there aren't any plugins, then the files aren't written to disk. The resulting
+/// file looks something like this (order of keys is not guaranteed):
+/// {
+///   "info": "This is a generated file; do not edit or check into version control.",
+///   "plugins": {
+///     "ios": [
+///       {
+///         "name": "test",
+///         "path": "test_path",
+///       }
+///     ],
+///     "android": [],
+///     "macos": [],
+///     "linux": [],
+///     "windows": [],
+///     "web": []
+///   },
+///   "dependencyGraph": [
```

Comment:
Is the plan to split "dependencyGraph" per platform in a separate PR?

Reply:
I don't think it makes much sense to split it anymore. It doesn't really add any value. I started that way, but then we'd have change the way the dependencies are handled in gradle, which at that point felt unnecessary. What do you think?

Reply:
Gradle is working around the fact that `dependencyGraph` isn't split per platform.  I think it would be ideal if Gradle didn't have to workaround that and instead just read this file. This could be done in a separate PR though if that makes sense.

Reply:
Makes sense, I did it here :)